### ISSUE TITLE
Thumbnails: I don't see any. What do I need to allow/open up to get them?

### ISSUE THREAD

**Issue by znoteer** [shaarli/Shaarli#1911]
The example shaare of the calm jazz music that comes with a new installation of shaarli has no image. I've seen the image on the demo sight, so I know one is missing.

Likewise, despite having tried to synchronize the thumbnails for my imported bookmarks, both via the "Configure your shaarli" page, and via the picture wall, I see zero images on my picture wall or in any of my shaares.

I suspect it's because I've got something turned off/set to restrictive/etc in some browser setting. Would a restrictive cookie, java-script or privacy setting hinder thumbnail fetching?

**Comment by nodiscc** [comment]
Hi,

- are there any errors in your PHP/webserver logs when fetching thumbnails or trying to display them?
- are there any errors in your browser console/network tab when fetching thumbnails or trying to display them?
- make sure required [PHP extensions](https://shaarli.readthedocs.io/en/master/Server-configuration/#php) are loaded

**Comment by znoteer** [comment]
>     * are there any errors in your PHP/webserver logs when fetching thumbnails or trying to display them?

None.

>     * are there any errors in your browser console/network tab when fetching thumbnails or trying to display them?

Oh, yeah, are there ever. 

Here are some warnings that are obviously related to the thumbnail sync operation:

```
13:05:25.318 Strict-Transport-Security: The connection to the site is untrustworthy, so the specified header was ignored.
[picture-wall](https://fbx.local/shaarli/picture-wall?)
13:05:30.047 Strict-Transport-Security: The connection to the site is untrustworthy, so the specified header was ignored.
[thumbnails](https://fbx.local/shaarli/admin/thumbnails)
13:05:30.049 Strict-Transport-Security: The connection to the site is untrustworthy, so the specified header was ignored.
[thumbnails](https://fbx.local/shaarli/admin/thumbnails)
13:05:30.156 Strict-Transport-Security: The connection to the site is untrustworthy, so the specified header was ignored.
[thumbnails_update.min.js](https://fbx.local/shaarli/tpl/default/js/thumbnails_update.min.js?v=8c6b4e5b8cd8ca1dc5c0fb4a018f862d2333231bab7f5904ee79006ce64707c6)
13:05:32.743 Strict-Transport-Security: The connection to the site is untrustworthy, so the specified header was ignored.
[update-thumbnail](https://fbx.local/shaarli/admin/shaare/2576/update-thumbnail)
```

I don't know if any of the other warnings are related and I can't figure out how to export the output of the console in FF to dump them all.

The are errors, but it is less obvious which might be related to thumbnails. Any clues about what to look for in the errors? like keywords?

**Comment by nodiscc** [comment]
> Strict-Transport-Security: The connection to the site is untrustworthy, so the specified header was ignored

Unrelated. This just tells you that the [Strict-Transport-Security header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security) was ignored since you are (probably?) using self-signed certificates. This has no impact.

> how to export the output of the console in FF to dump them all.

- Open developer tools (F12), open the `Console` tab
- Make sure logging is enabled for all events at the top right side of the console (`Errors` `Warnings` `...` `Requests`)
- Navigate to your Shaarli instance where the thumbnails should be displayed
- Right click anywhere in the console > `Copy all messages`
- Paste contents of the console here

Have you checked that required [PHP extensions](https://shaarli.readthedocs.io/en/master/Server-configuration/#php) are loaded?

**Comment by nodiscc** [comment]
The problem looks specific to the Freedombox packaging/installation, so you should report this on their bug tracker. Currently we only support installation methods described on https://shaarli.readthedocs.io/en/master/Installation/ (thumbnails are known to work with these methods if you follow them to the letter). 

There are too many variables in the Freedombox setup we don't know about (proxy/PHP configuration, installation paths, permissions, postinst script..). It will also help fixing the problem for other FB users. Once the Freedombox package works well enough, we could list it [here](https://shaarli.readthedocs.io/en/master/Community-and-related-software/#integration-with-other-platforms), but ultimately fixing such installation bugs is the job of the packager/distributor.

As a last attempt:
- please check that thumbnails are actually enabled under `Tools > Configure your Shaarli > Enable thumbnails`
- please post the output of

```
sudo ls -ld /var/lib/shaarli/data /var/cache/shaarli/cache /var/cache/shaarli/pagecache /var/cache/shaarli/tmp
sudo ls -l /var/cache/shaarli/cache
```

> are there any errors in your PHP/webserver logs when fetching thumbnails or trying to display them?
>> None.

- This surprises me, can you please go to `Tools > Configure your Shaarli`, click `Synchronize thumbnails`, tell us what happens, and post a full copy of `/var/log/apache2/error.log` ?

**Comment by znoteer** [comment]
Yes, thumbnails is set to "All" under "Configure your Shaarli".

```
fbx@fbx:~$ ls -ldF /var/lib/shaarli/data/ /var/cache/shaarli/cache/ /var/cache/shaarli/pagecache/ /var/cache/shaarli/tmp
drwxr-xr-x 2 www-data www-data 4096 Aug 22 08:16 /var/cache/shaarli/cache/
drwxr-xr-x 2 www-data www-data 4096 Aug 22 08:16 /var/cache/shaarli/pagecache/
drwxr-xr-x 2 www-data www-data 4096 Nov 13 17:14 /var/cache/shaarli/tmp/
drwxr-xr-x 2 www-data www-data 4096 Oct  3 19:41 /var/lib/shaarli/data/
fbx@fbx:~$ ls -lF /var/cache/shaarli/cache/
total 0
fbx@fbx:~$ ls -lAF /var/cache/shaarli/cache
total 0
fbx@fbx:~$ sudo tail -F /var/log/apache2/error.log
[sudo] password for saf: 

```

I ran the above tail -F command on the server before launching the sync'ing operation, so what you see above, after the tail command is the full output of the apache2 error.log file.

Here are some more indications that nothing is in the error (or other) log(s):

```
saf@fbx:~$ sudo su -
root@fbx:~# ls -lAF /var/log/apache2/*.log
-rw-r----- 1 root adm 0 Sep  8 00:00 /var/log/apache2/access.log
-rw-r----- 1 root adm 0 Sep  8 00:00 /var/log/apache2/error.log
-rw-r----- 1 root adm 0 Oct 23 00:00 /var/log/apache2/other_vhosts_access.log
-rw-r----- 1 root adm 0 Sep 19  2021 /var/log/apache2/ssl_access.log
root@fbx:~# 
```

As for what happens when I click "Synchronise", it's this sort of thing: 

![Screenshot-shaarli-sync-thumbnails_2022-11-21_20-58-42](https://user-images.githubusercontent.com/10201567/203199015-e6d3d2d8-9d26-44ac-b3a8-a71195e8fa88.png)

That is too say, it looks like things are progressing normally as it runs through all my shaares.

I'll wait for your reaction to this comment before creating an issue with FBX.

**Comment by ArthurHoaro** [comment]
You could try to set the setting `dev.debug` in your `config.json.php` file to `true` to get more logs about the thumbnail retrieval attempts. Note that I just fixed a bug with the debug mode and PHP 8 on develop, so it might not work properly depending on your versions of PHP and Shaarli.

With this setting, you should normally see these type of warnings in the server logs:

> WebThumbnailer\Exception\ThumbnailNotFoundException: No thumbnail could be found using default finder ...

Let's see if you get anything else.

Note that on your last screenshot, you are supposed to see the thumbnails being retrieved, for example:
![image](https://user-images.githubusercontent.com/1962678/204024517-3fdd34d1-6047-4527-9c1a-935263632d9e.png)

**Comment by znoteer** [comment]
@ArthurHoaro 

Shaarli 0.12.1
PHP 7.4

I can't find "dev.debug" in order to change it to "true":

```
saf@fbx:~$ ls -lAF /var/lib/shaarli/data/
total 1320
-rw-r--r-- 1 www-data www-data   2359 Nov 25 15:01 config.json.php
-rw-r--r-- 1 www-data www-data   2338 Oct 12 06:08 config.json.php_orig
-rw-r--r-- 1 www-data www-data 395083 Nov 25 15:29 datastore.php
-rw-r--r-- 1 www-data www-data 217455 Nov 25 15:29 history.php
-rw-r--r-- 1 www-data www-data 704970 Oct  3 19:43 import.2022-10-03.log
-rw-r--r-- 1 www-data www-data     67 Nov 18 06:58 ipbans.php
-rw-r--r-- 1 www-data www-data      6 Nov 25 15:04 lastupdatecheck.txt
-rw-r--r-- 1 www-data www-data   1464 Nov 18 06:58 log.txt
-rw-r--r-- 1 www-data www-data     64 Oct  1 19:53 updates.txt
saf@fbx:~$ grep -e "dev.debug" /var/lib/shaarli/data/*
saf@fbx:~$ grep -e "dev\.debug" /var/lib/shaarli/data/*
saf@fbx:~$ 
```

Maybe "dev.debug" doesn't exist for my version of either PHP or Shaarli?

**Comment by nodiscc** [comment]
@znoteer you need to create the key as documented [here](https://shaarli.readthedocs.io/en/master/Shaarli-configuration/#configuration-file-example)

**Comment by znoteer** [comment]
I added the dev.debug key,  restarted apache and php and ran the thumbnails syncro routine. I've truncated the file because the tail end is nothing more than many repetitions of the last line I left in the file, one for each shaare.

TL;DR no new information is revealed :/

[shaarli-thumbnail_console-export-2022-11-29_10-55-23_truncated.txt](https://github.com/shaarli/Shaarli/files/10114839/shaarli-thumbnail_console-export-2022-11-29_10-55-23_truncated.txt)

Should I report this to Freedombox now? Or do you want to try something else, first?

**Comment by nodiscc** [comment]
You will find debug logs in your server logs, not your browser console.

But yes, you should report this to Freedombox with detailed steps on how to reproduce the problem. This looks like an installation/server configuration problem.

Ping @jvalleroy who maintains the Debian/Fbx package.

**Comment by znoteer** [comment]
Oops :) Just checked the server logs and nothing has been written to the apache logs since 7 September (access.log, error.log) and the only thing in the php log about my recent run was about restarting PHP after I editted the config.json the file, being ready to handle connections and the systemd monitor being set to an interval of 1000ms.

I'll open an issue with Freedombox. Thanks for helping debug.

**Comment by jvalleroy** [comment]
This issue has been fixed in the package version 0.12.1+dfsg-8, in Debian testing and bullseye-backports. Thanks for reporting it!
